data-protocols/ids/ids-api-transfer/src/main/java/org/eclipse/dataspaceconnector/ids/api/transfer/ArtifactRequestController.java

```diff
@@ -112,17 +113,26 @@ public Response request(ArtifactRequestMessage message) {
 
 
         // TODO this needs to be deserialized from the artifact request message
-        var destinationMap = (Map<String, Object>) message.getProperties().get(ArtifactRequestController.DESTINATION_KEY);
+        Map<String, Object> messageProperties = message.getProperties();
+        var destinationMap = (Map<String, Object>) messageProperties.get(ArtifactRequestController.DESTINATION_KEY);
         var type = (String) destinationMap.get("type");
 
-        Map<String, String> properties = (Map<String, String>) destinationMap.get("properties");
+        Map<String, String> destinationProperties = (Map<String, String>) destinationMap.get("properties");
```

Comment:
is it possible that `destinationMap.get("properties")` returns null in some situation ? May be we can getOrDefault with and empty map.